src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts

```diff
@@ -65,47 +66,50 @@ export class TerminalLinkManager extends DisposableStore {
-			const validatedProvider = this._instantiationService.createInstance(TerminalValidatedLocalLinkProvider,
-				this._xterm,
-				this._processManager.os || OS,
-				wrappedTextLinkActivateCallback,
-				this._wrapLinkHandler.bind(this),
-				this._tooltipCallback.bind(this),
-				async (linkCandidates, cb) => {
-					for (const link of linkCandidates) {
-						const result = await this._resolvePath(link);
-						if (result) {
-							return cb(result);
-						}
-					}
-					return cb(undefined);
-				});
-			this._standardLinkProviders.set(TerminalValidatedLocalLinkProvider.id, validatedProvider);
+			this._setupLinkDetector(TerminalLocalLinkDetector.id, this._instantiationService.createInstance(TerminalLocalLinkDetector, this._xterm, this._processManager.os || OS, this._resolvePath.bind(this)));
 		}
+		this._setupLinkDetector(TerminalWordLinkDetector.id, this._instantiationService.createInstance(TerminalWordLinkDetector, this._xterm));
 
-		this._capabilities.get(TerminalCapability.CwdDetection)?.onDidChangeCwd(cwd => {
+		capabilities.get(TerminalCapability.CwdDetection)?.onDidChangeCwd(cwd => {
 			this.processCwd = cwd;
 		});
 
-		// Word links
-		const wordProvider = this._instantiationService.createInstance(TerminalWordLinkProvider, this._xtermTerminal, this._capabilities, this._wrapLinkHandler.bind(this), this._tooltipCallback.bind(this));
-		this._standardLinkProviders.set(TerminalWordLinkProvider.id, wordProvider);
+		// Setup link openers
+		this._openers.set(TerminalLinkType.LocalFile, this._instantiationService.createInstance(TerminalLocalFileLinkOpener, this._processManager.os || OS));
+		this._openers.set(TerminalLinkType.LocalFolderInWorkspace, this._instantiationService.createInstance(TerminalLocalFolderInWorkspaceLinkOpener));
+		this._openers.set(TerminalLinkType.LocalFolderOutsideWorkspace, this._instantiationService.createInstance(TerminalLocalFolderOutsideWorkspaceLinkOpener));
+		this._openers.set(TerminalLinkType.Search, this._instantiationService.createInstance(TerminalSearchLinkOpener, capabilities));
+		this._openers.set(TerminalLinkType.Url, this._instantiationService.createInstance(TerminalUrlLinkOpener, !!this._processManager.remoteAuthority));
+
+		// TODO: Hook up wrap link handler which is what gets modifier down working
 
 		this._registerStandardLinkProviders();
 	}
 
+	private _setupLinkDetector(id: string, detector: ITerminalLinkDetector): void {
+		const wordProvider = this._instantiationService.createInstance(TerminalLinkDetectorAdapter, detector);
+		wordProvider.onDidActivateLink(e => this._openLink(e));
+		wordProvider.onDidShowHover(e => this._tooltipCallback(e.link, e.viewportRange, e.modifierDownCallback, e.modifierUpCallback));
+		this._standardLinkProviders.set(id, wordProvider);
+	}
+
+	private async _openLink(link: ITerminalSimpleLink): Promise<void> {
+		console.log('open', link);
+		await this._openers.get(link.type)?.open(link);
```

Comment:
maybe we should throw if the opener is undefined?

Reply:
I was considering that, I guess it probably is the right thing to do.